File: Bcore/src/main/java/top/niunaijun/blackbox/http/ServerManager.java
package top.niunaijun.blackbox.http;

import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.yanzhenjie.andserver.AndServer;
import com.yanzhenjie.andserver.Server;

import java.util.concurrent.TimeUnit;

import top.niunaijun.blackbox.BlackBoxCore;
import top.niunaijun.blackbox.touch.ControlThread;

public class ServerManager {

    private static Server mServer;
    public static Server get()
    {
        if(mServer == null)
        {
            Context context = BlackBoxCore.getContext();
            mServer = AndServer.webServer(context)
                    .port(8080)
                    .timeout(10, TimeUnit.SECONDS)
                    .listener(new Server.ServerListener() {
                        @Override
                        public void onStarted() {
                            // TODO The server started successfully.
                            ControlThread.get().start();
                            Toast.makeText(context, "web start", Toast.LENGTH_LONG).show();
                        }

                        @Override
                        public void onStopped() {
                            // TODO The server has stopped.
                        }

                        @Override
                        public void onException(Exception e) {
                            // TODO An exception occurred while the server was starting.
                        }
                    })
                    .build();
        }
        return mServer;
    }

    /**
     * Start server.
     */
    public static void startServer() {
        Server server = get();
        if (server.isRunning()) {
            // TODO The server is already up.
        } else {
            server.startup();
        }
    }

    /**
     * Stop server.
     */
    public static void stopServer() {
        Server server = get();
        if (server.isRunning()) {
            server.shutdown();
        } else {
            Log.w("AndServer", "The server has not started yet.");
        }
    }
}

File: app/src/main/java/top/niunaijun/blackboxa/http/ServerManager.kt
package top.niunaijun.blackboxa.http

import android.content.Context
import android.util.Log
import android.widget.Toast
import com.yanzhenjie.andserver.AndServer
import com.yanzhenjie.andserver.Server
import com.yanzhenjie.andserver.Server.ServerListener
import java.util.concurrent.TimeUnit


class ServerManager(context: Context?) {
    private val mServer: Server = AndServer.webServer(context!!)
        .port(8080)
        .timeout(10, TimeUnit.SECONDS)
        .listener(object : ServerListener {
            override fun onStarted() {
                // TODO The server started successfully.
                Toast.makeText(context, "web start", Toast.LENGTH_LONG).show()
            }

            override fun onStopped() {
                // TODO The server has stopped.
                Toast.makeText(context, "web stop", Toast.LENGTH_LONG).show()
            }

            override fun onException(e: Exception) {
                // TODO An exception occurred while the server was starting.
            }
        })
        .build()

    /**
     * Start server.
     */
    fun startServer() {
        if (mServer.isRunning) {
            // TODO The server is already up.
        } else {
            mServer.startup()
        }
    }

    /**
     * Stop server.
     */
    fun stopServer() {
        if (mServer.isRunning) {
            mServer.shutdown()
        } else {
            Log.w("AndServer", "The server has not started yet.")
        }
    }
}